### DISCUSSION TITLE
Fp-Growth

### DISCUSSION THREAD

**Discussion by AlbertAssaad** [rasbt/mlxtend#906] [Q&A] [answered]
Hi, I made a python program to get FP-Growth to a huge amount of transactions using your library 
Is it normal that the result has redundant swapped items? example:

antecedents | consequents | conviction
-- | -- | --
Item X | Item Y | 1.01198577066718
Item Y | Item X | 1.01064124141153

It's doing this for most first lines...
Thank you

**Comment by rasbt** [comment, marked as answer]
That's a good question. Yes, this is intentional. One of the main reasons is that it is hard to say which one someone is interested in. Also, it affects certain metrics (like conviction). Here's a screenshot from the docs  athttp://rasbt.github.io/mlxtend/user_guide/frequent_patterns/association_rules/
![Screen Shot 2022-04-11 at 12 20 17 PM](https://user-images.githubusercontent.com/5618407/162795231-92f3a280-e5a5-4bc1-8156-4e762d9ef5b9.png)
